### PR TITLE
[net9.0] Update with main changes

### PR DESCRIPTION
**Pull request by rmarinho**
### Description of Change

Update with latest ci changes and fixes from main

### CODE REVIEW

File: src/TestUtils/src/Microsoft.Maui.IntegrationTests/TemplateTests.cs [hunk 1]
```diff
@@ -113,6 +113,55 @@ public void BuildUnpackaged(string id, string framework, string config)
...
+			extendedBuildProps.Add("PublishAot=true");
+			extendedBuildProps.Add("PublishAotUsingRuntimePack=true");  // TODO: This parameter will become obsolete https://github.com/dotnet/runtime/issues/87060
+
+			Assert.IsTrue(DotnetInternal.Publish(projectFile, "Release", framework: framework, properties: extendedBuildProps, runtimeIdentifier: runtimeIdentifier),
+				$"Project {Path.GetFileName(projectFile)} failed to build. Check test output/attachments for errors.");
+		}
+
+		[Test]
+		[TestCase("maui", $"{DotNetCurrent}-ios", "ios-arm64")]
+		public void PublishNativeAOTCheckWarnings(string id, string framework, string runtimeIdentifier)
```
Comment (rmarinho) [inline src/TestUtils/src/Microsoft.Maui.IntegrationTests/TemplateTests.cs:139]: @ivanpovazan seems this is failing on this branch, for net9.0 can you take a look ?

Comment (rmarinho) [reply src/TestUtils/src/Microsoft.Maui.IntegrationTests/TemplateTests.cs:139]: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=8817979&view=ms.vss-test-web.build-test-results-tab&runId=90608599&resultId=100050&paneView=debug

Expected warnings file path 'ILC' was not found.
Expected: not null
But was:  null

File: src/TestUtils/src/Microsoft.Maui.IntegrationTests/Utilities/BuildWarningsUtilities.cs [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+					new WarningsPerCode
+					{
+						Code = "IL3050",
+						Messages = new List<string>
+						{
+							"<Module>..cctor(): Using member 'System.Enum.GetValues(Type)' which has 'RequiresDynamicCodeAttribute' can break functionality when AOT compiling. It might not be possible to create an array of the enum type at runtime. Use the GetValues<TEnum> overload or the GetValuesAsUnderlyingType method instead.",
+						}
+					},
+				}
+			},
```
Comment (ivanpovazan) [inline src/TestUtils/src/Microsoft.Maui.IntegrationTests/Utilities/BuildWarningsUtilities.cs:140]: @rmarinho This warning got fixed with: https://github.com/xamarin/xamarin-macios/pull/19516
And should be removed from the expected list of warnings. 

Do you want me to remove/test it, or you want to give it a go (basically selected lines should be deleted)?

File: src/TestUtils/src/Microsoft.Maui.IntegrationTests/TemplateTests.cs (hunk 1 above)
Comment (ivanpovazan) [reply src/TestUtils/src/Microsoft.Maui.IntegrationTests/TemplateTests.cs:139]: Please see my comment at: https://github.com/dotnet/maui/pull/19441/files/f7ac0366ac299b1486b26ebada4a51b66a3d846d#r1428262698

**Pull request merged**